backend/src/routing/controllers/api/Abitur.ts

```diff
@@ -11,9 +12,44 @@ export default class Abitur {
     static POSTapplyForTopic(req: Request, res: Response): void {
         if (rejectWhenValidationsFail(req, res)) return;
 
-        const {responsibleTeacherId, art, bezugsfachId, partnerStudentId, referenzfachId, topic} = req.body;
-        const sql = 'INSERT INTO abiturpruefungen (betreuendeLehrkraftID, art, bezugsfachID, partnerStudentId, referenzfachID, thema) VALUES (?,?,?,?,?,?)';
-        insertData(sql, [responsibleTeacherId, art, bezugsfachId, partnerStudentId, referenzfachId, topic], defaultInsertCallback(res));
+        let sql = 'SELECT id FROM abiturpruefungen WHERE studentID = ?';
+        getAllResults(sql, [getStudentId()], (a, err) => {
+            if (err) {
+                res.status(500).json(err.name);
+            } else {
+                if (a.length > 0) {
+                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                    // @ts-ignore
+                    updateUpdateColumns(a[0].id);
+                } else {
+                    insertFirstSubmission();
+                }
+            }
+        });
+
+        const updateUpdateColumns = (id: number) => {
+            const {updatedExaminer, updatedBezugsfach, updatedPartnerStudentName, updatedReferenzfach, updatedTopicArea, problemQuestion, updatedProblemQuestion, presentationForm, updatedPresentationForm} = req.body;
+            let setString = '';
+            const args = [];
+            setString += ' updatedExaminer = ?'; args.push(updatedExaminer ? updatedExaminer : null);
+            setString += ', updatedBezugsfach = ?'; args.push(updatedBezugsfach ? updatedBezugsfach : null);
+            setString += ', updatedPartnerStudentName = ?'; args.push(updatedPartnerStudentName ? updatedPartnerStudentName : null);
+            setString += ', updatedReferenzfach = ?'; args.push(updatedReferenzfach ? updatedReferenzfach : null);
+            setString += ', updatedThema = ?'; args.push(updatedTopicArea ? updatedTopicArea : null);
+            setString += ', updatedProblemQuestion = ?'; args.push(updatedProblemQuestion ? updatedProblemQuestion : null);
+            setString += ', updatedPresentationForm = ?'; args.push(updatedPresentationForm ? updatedPresentationForm : null);
+            if (problemQuestion) { setString += ', problemQuestion = ?'; args.push(problemQuestion); }
+            if (presentationForm) { setString += ', presentationForm = ?'; args.push(presentationForm); }
+            sql = 'UPDATE abiturpruefungen SET' + setString +' WHERE id = ?';
+            args.push(id);
+            updateData(sql, args, defaultUpdateCallback(res));
+        };
+
+        const insertFirstSubmission = () => {
+            const {examiner, examType, bezugsfach, partnerStudentName, referenzfach, topicArea} = req.body;
+            sql = 'INSERT INTO abiturpruefungen (examiner, art, bezugsfach, partnerStudentName, referenzfach, thema, studentID) VALUES (?,?,?,?,?,?,?)';
```

Comment:
Wieso werden oben die undefineds zu null gemapped und hier nicht?

Reply:
Weil diese Anfrage hier nur ein mal geschickt wird. Und dabei alle diese Werte geschickt werden vom Frontend. Bei dem "zweiten Teil" der anfrage kann es vorkommen dass diese öfter als ein mal geschickt wird. Und es ist wichtig, dass ein UpdatedValue wieder genullt wird, wenn es beim zweiten mal nicht mitgeschickt wird. 

Der pdf generator wird drauf schauen, ob ein updated value da ist und wenn eins da ist das updated value in die pdf schreiben. Deshalb ist es bei den updated values wichtig, dass sie genullt werden wenn sie nicht geschickt werden